db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AlterTableNode.java

```diff
@@ -524,22 +524,6 @@ public ConstantAction	makeConstantAction() throws StandardException
 
             tableElementList.genConstraintActions(false, conActions, getRelativeName(), schemaDescriptor,
                                                   getDataDictionary());
-
-            for (ConstantAction cca : conActions) {
-                if (getGenericConstantActionFactory().primaryKeyConstantActionCheck(cca)) {
-                    DataDictionary dd = getDataDictionary();
-                    // Check to see if a constraint of the same type
-                    // already exists
-                    ConstraintDescriptorList cdl =
-                        dd.getConstraintDescriptors(baseTable);
-
-                    if (cdl.getPrimaryKey() != null) {
-                        throw StandardException.newException(
-                            SQLState.LANG_ADD_PRIMARY_KEY_FAILED1,
-                            baseTable.getQualifiedName());
-                    }
```

Comment:
why has this been removed? looks like a check, is that check superflous?

Reply:
primaryKeyConstantActionCheck was checking if cca was of type ConstraintConstantAction, but that class was never used.

Reply:
aaaahhh!